### ISSUE TITLE
[Bug]: After max iterations are hit, they do not reset on a user message

### ISSUE THREAD

**Issue by neubig** [OpenDevin/OpenDevin#3323]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

I had a long and productive OpenDevin session, and hit the maximum number of iterations 100. I wanted to continue the session by providing additional input, but it seems that `max_iterations` is not extended on additional user input.

![Screenshot 2024-08-09 at 11 31 20 AM](https://github.com/user-attachments/assets/0937a76b-9b5a-40bb-b891-9b825f05d839)

It would be nice to be able to continue long sessions.

### Current OpenDevin version

```bash
commit 5e6fd58f56840f20fc619349428092b7ddae45fc
```

### Installation and Configuration

```bash
make build; make run
```

### Model and Agent

- Model: Claude 3.5 Sonnet
- Agent: CodeActAgent

### Reproduction Steps

1. Run OpenDevin for a long time
2. Type "continue" when you get a max iterations error

### Logs, Errors, Screenshots, and Additional Context

Related to https://github.com/OpenDevin/OpenDevin/issues/2710 , but a little different.

**Comment by li-boxuan** [comment]
It actually does continue (for one more step), as shown in the screenshot below:

<img width="426" alt="Screenshot 2024-08-11 at 10 51 07 AM" src="https://github.com/user-attachments/assets/22441aa4-eaf1-4cee-9b13-97fd860c023b">

You just need to keep hitting the "continue" button.

Some UI design might be needed if we want to give users more flexibility, e.g. resetting the limit, increasing the limit exponentially, etc.

**Comment by neubig** [comment]
Yeah, my expectation would be that after the user gave input it would reset. But I'm not sure what others think.

**Comment by enyst** [comment]
Now that I think about it, `max_iterations` is much more useful when running headless and with user confirmation fully disabled, than in interactive UI runs. Particularly when the user will have the ability to stop or pause easily, it might become unnecessary to restrict iterations? Alternatively, a reset sounds good to me.

**Comment by neubig** [comment]
`max_iterations` is still a little bit useful to me in interactive ui runs because I will sometimes start one running, then go to the grocery store or something, and then come back. But I do think it'd be good to be able to continue for another 100 turns or so if you send a new message.

**Comment by tofarr** [comment]
This should now be reaolved
